andela/caret-bn-backend
# Ft social media signin 168781680

#### What does this PR do?
- Integrates Social Media Authentication into the application
#### Description of Task to be completed?

- Allow users to authenticate their barefoot nomad accounts using Google
- Allow users to authenticate their barefoot nomad accounts using Facebook

#### How should this be manually tested?

1. Clone this repo

2. Run ` npm install ` to install all npm packages

3. Go to the following url https://developers.google.com/oauthplayground/ and authorize the following APIs (This is done in step 1 of the Playground process).
 a) Google OAuth apiv2
      - https://www.googleapis.com/auth/userinfo.email
      - https://www.googleapis.com/auth/userinfo.profile

4. Exchange the authorization code for tokens (This is done in step 2 of the Playground process)

5. Go to https://developers.facebook.com/ create a new app called barefoot_nomad

6. Go to  https://developers.facebook.com/tools/access_tokens and create access tokens for your applications

7. Add the following environment variables to your application 

```
GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=
GOOGLE_ACCESS_TOKEN=

FACEBOOK_CLIENT_ID=
FACEBOOK_CLIENT_SECRET=
FACEBOOK_ACCESS_TOKEN=
```
8. Start up your application

9. Using postman make a ` POST ` request to the following endpoints

      a. ` localhost:<port>/api/v1/auth/facebook/ `
      b. ` localhost:<port>/api/v1/auth/google/ `

10. Attempt to change the access_tokens to see proper error message display


#### Any background context you want to provide?

N / A

#### What are the relevant pivotal tracker stories?
##### #168781680
#### Screenshots (if appropriate)

<img width="746" alt="Screen Shot 2019-10-07 at 16 38 11" src="https://user-images.githubusercontent.com/13886438/66634798-1eaf1a00-ec0e-11e9-8e79-b29a305ecc71.png">`

<img width="1125" alt="Screen Shot 2019-10-08 at 09 15 53" src="https://user-images.githubusercontent.com/13886438/66634863-443c2380-ec0e-11e9-9620-ca1e0873ee7d.png">
<img width="1020" alt="Screen Shot 2019-10-11 at 10 03 22" src="https://user-images.githubusercontent.com/13886438/66635067-b6146d00-ec0e-11e9-9a5f-6d8f35e07eab.png">
<img width="1373" alt="Screen Shot 2019-10-11 at 10 03 31" src="https://user-images.githubusercontent.com/13886438/66635068-b6146d00-ec0e-11e9-8f5f-fb4a04e56416.png">
<img width="1370" alt="Screen Shot 2019-10-11 at 10 03 35" src="https://user-images.githubusercontent.com/13886438/66635071-b6146d00-ec0e-11e9-8cd3-d7c8bb71a06d.png">
<img width="1370" alt="Screen Shot 2019-10-11 at 10 03 58" src="https://user-images.githubusercontent.com/13886438/66635072-b6ad0380-ec0e-11e9-9e28-dcda60814744.png">
<img width="1358" alt="Screen Shot 2019-10-11 at 10 04 15" src="https://user-images.githubusercontent.com/13886438/66635073-b6ad0380-ec0e-11e9-98de-90543bde02fc.png">
<img width="1361" alt="Screen Shot 2019-10-11 at 10 05 12" src="https://user-images.githubusercontent.com/13886438/66635074-b6ad0380-ec0e-11e9-86a2-7d5959294480.png">








#### Questions:
N/A
